application/classes/Model/User.php

```diff
@@ -192,14 +192,18 @@ public static function all() {
 		return (new Model_User())->find_all();
 	}
 	
+	public static function valid() {
+		return (new Model_user())->where('email', 'NOT LIKE', 'customer%')->find_all();
+	}
+	
 	/**
 	 * Retrieve all convention registrants (i.e. users who registered for a convention)
 	 * @param Model_Convention $con Convention to retrieve users for
 	 * @return Generator list of users
 	 */
 	public static function byConvention(Model_Convention $con) {
 		$seen = [];
-		foreach ((new Model_Ticket())->with('sale')->where('convention_id', '=', $con->pk())->find_all() as $ticket) {
+		foreach ((new Model_Sale())->where('convention_id', '=', $con->pk())->find_all() as $ticket) {
```

Comment:
I'm not sure I follow - its a standard `where('field','=',$id)` pattern... I don't like the way Kohana implements `where()`, that you have to specify the operator as a string, but I have a feeling that doing it another way may be worse.